src/player/Player.php

```diff
@@ -865,6 +865,29 @@ static function() : void{
 		Timings::$playerChunkSend->stopTiming();
 	}
 
+	/**
+	 * Checks if the player is currently on the ground. This is more accurate than {@link Player::isOnGround()} but slower.
+	 */
+	public function isActuallyOnGround() : bool{
```

Comment:
Yeah, I'm not a fan of this naming either. Also, why do we actually need this?

Reply:
> Yeah, I'm not a fan of this naming either. Also, why do we actually need this?

We need it because when using the normal is on ground method the break progress indicator is still too fast when jumping

Reply:
Sounds like that warrants investigation into regular onGround behaviour.

Reply:
Legend has it that `PlayerAuthInputPacket` should allow making onGround checks more accurate, so this hack wouldn't be needed anymore.

Reply:
I think the PR should use the built-in onGround for now. It doesn't make sense to add an extra function specifically for this.

Once the proper onGround is fixed per #6547 we won't need this workaround. I don't like adding stuff to the API that we know we're gonna delete anyway.